### DISCUSSION TITLE
Trying to use nested objects when extending the theme

### DISCUSSION THREAD

**Discussion by ChangoMan** [tailwindlabs/tailwindcss#9245] [Help] [answered]
Hello! My question is about using nested objects when extending a theme. This seems to work on the `backgroundColor` and `color` utilities, but it doesn't work for any other utility. Is this because the `color` utility is special?

For Example, this works with `backgroundColor`:

```
// config
module.exports = {
  theme: {
    extend: {
      backgroundColor: {
        button: {
          primary: '#000000'
        }
      },
    }
  }
}

// Works with
@apply bg-button-primary;
```

But this doesn't work with something like `borderRadius`:

```
// config
module.exports = {
  theme: {
    extend: {
      borderRadius: {
        button: {
          primary: '20px',
        },
      },
    }
  }
}

// This doesn't work with
@apply rounded-button-primary;
```

Is this expected, and if so, is there a way to get the nesting to work in the `borderRadius` example?

Thanks in advanced for any feedback!

**Comment by adamwathan** [comment, marked as answer]
Hey! We only automatically combine keys like this for colors, not for anything else. You'll have to just do this:

```js
module.exports = {
  theme: {
    extend: {
      borderRadius: {
        'button-primary': '20px',
      },
    }
  }
}
```

Could also write a little JS helper function yourself to combine the keys for you:

```js
function concatKeys(obj) {
  // ...
}

module.exports = {
  theme: {
    extend: {
      borderRadius: concatKeys({
        button: {
          primary: '20px',
        },
      }),
    }
  }
}
```
